app/Http/Controllers/Setup/Controller.php

```diff
@@ -125,17 +141,33 @@ public function save_auth_settings() {
     public function save_config() {
         $config = file_get_contents(__DIR__.'/../../../../.env.example');
 
-        // Generate an APP_KEY
-        $app_key = 'base64:'.base64_encode(Encrypter::generateKey('AES-256-CBC'));
-        self::write_config_value($config, 'APP_KEY', $app_key);
-
-        self::write_config_value($config, 'APP_DEBUG', 'false');
-        self::write_config_value($config, 'APP_ENV', 'production');
+        // Pull from existing ENV
+        foreach([
+            "APP_URL", "APP_DEBUG", "APP_ENV", "APP_KEY",
+            "ENABLE_WEBMENTION_RESPONSES", "ENABLE_TICKET_URL", "SHOW_RSVPS_IN_ICS",
+            "CACHE_DRIVER", "FILESYSTEM_DRIVER", "LOG_CHANNEL",
+            "QUEUE_CONNECTION", "SESSION_DRIVER", "SESSION_LIFETIME",
+            "LOGO_URL", "LOGO_WIDTH", "LOGO_HEIGHT", "FAVICON_URL",
+            "AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_DEFAULT_REGION", "AWS_BUCKET",
+            "AWS_URL",
+        ] as $k) {
+            if(!empty(env($k))) {
+                self::write_config_value($config, $k, env($k));
+            }
+        }
 
-        self::write_config_value($config, 'DB_HOST', session('setup.db_host'));
-        self::write_config_value($config, 'DB_DATABASE', session('setup.db_name'));
-        self::write_config_value($config, 'DB_USERNAME', session('setup.db_username'));
-        self::write_config_value($config, 'DB_PASSWORD', session('setup.db_password'));
+        // Generate an APP_KEY if needed
+        if($_ENV['APP_KEY'] == 'base64:v7ZDOfJbqzXdbbJ/3GYSAP+B4jm3rMlrWiNutsaQYEE=') {
+            $app_key = 'base64:'.base64_encode(Encrypter::generateKey('AES-256-CBC'));
+            self::write_config_value($config, 'APP_KEY', $app_key);
+        }
+        
+        if(empty(env('DATABASE_URL'))) {
```

Comment:
This should check whether `session('setup.db_host')` exists instead.

Reply:
In that case it should loop and check all of them. I decided to take a single piece of instrumentation which overrides all of these and have the setup omit them (as the ENV would be setup) in that case.